src/detectAdBlocker.ts

```diff
@@ -0,0 +1,65 @@
+/**
+Detect whether or not the user has an ad blocking extension enabled.
+A few ad blockers are not detectable with this approach e.g. Safari / Adblock
+Code inspired by just-detect-adblock's: https://git.io/JgL4L
+*/
+/*istanbul ignore file -- adElementBlocked can't be tested without patching each of the properties of
+HTMLElement.prototype that it accesses, defeating the purpose of the test! */
```

Comment:
See https://github.com/jsdom/jsdom/issues/1261
Is anyone aware of a workaround for this so that this module can be meaningfully tested with jest? 
Or would we need end-to-end tests in order to properly test this functionality?

Reply:
You could simply test that removing the ad element between two `window.requestAnimationFrame` calls results in a `true` response.

Reply:
I thought about doing something like this but then wasn't really sure that the test would provide much value. 
Removing the element from the document wouldn't be very realistic as this isn't what the ad blockers really do. And there's a risk that that sort of test would provide a false sense of security if someone breaks `adElementBlocked` in the future but tests pass.
But happy to give that a try if there's something I'm missing.